tests/models/test_dependencies_schema.py

```diff
@@ -197,9 +196,16 @@ def test_multiple_set_retriever_schema_with_same_name():
         doc_uri="doc-uri-3",
         other_columns=["column1", "column2"],
     )
+    set_retriever_schema(
+        name="my_ret_2",
+        primary_key="primary-key",
+        text_column="text-column",
+        doc_uri="doc-uri",
+        other_columns=["column1", "column2"],
+    )
 
-    with pytest.raises(
-        MlflowException, match=r"A retriever schema with the name 'my_ret_1' already exists."
+    with pytest.warns(
+        UserWarning, match=r"A retriever schema with the name 'my_ret_1' already exists."
```

Comment:
Can we augment this message to specify what the behavior is in this scenario? Also sounds like before we supported passing `name`, the second retriever schema would take precedence rather than the first - we probably should have the same behavior here (?)

Reply:
Updated the warning :)